src/service/future.rs

```diff
@@ -50,3 +55,40 @@ where
         }
     }
 }
+
+pin_project! {
+    /// Response future for [`crate::handler::OnMethod`] and
+    /// [`crate::service::OnMethod`].
+    pub struct OnMethodResponseFuture<S, F, B>
+    where
+        S: Service<Request<B>>,
+        F: Service<Request<B>>
+    {
+        #[pin]
+        pub(crate) f: RouteFuture<S, F, B>,
+        pub(crate) req_method: Method,
+    }
+}
+
+impl<S, F, B> Future for OnMethodResponseFuture<S, F, B>
+where
+    S: Service<Request<B>, Response = Response<BoxBody>>,
+    F: Service<Request<B>, Response = Response<BoxBody>, Error = S::Error>,
+{
+    type Output = Result<Response<BoxBody>, S::Error>;
+
+    fn poll(self: Pin<&mut Self>, cx: &mut Context<'_>) -> Poll<Self::Output> {
+        let this = self.project();
+        let mut res: Response<_> = ready!(this.f.poll(cx)?);
+
+        // HEAD must not contain a body according to
```

Comment:
> A response to a HEAD method should not have a body. If it has one anyway, that body must be ignored: any representation headers that might describe the erroneous body are instead assumed to describe the response which a similar GET request would have received.

see: [mdn](https://developer.mozilla.org/en-US/docs/Web/HTTP/Methods/HEAD)

It is useful for request part of a document that the server should return. 
For file download with Range header, it tells how large the document is.

Sorry my english is poor.

Reply:
Good catch! Fixed in 4c1a1db6d69c8